akd_local_auditor/src/storage/s3.rs

```diff
@@ -0,0 +1,180 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+//
+// This source code is licensed under both the MIT license found in the
+// LICENSE-MIT file in the root directory of this source tree and the Apache
+// License, Version 2.0 found in the LICENSE-APACHE file in the root directory
+// of this source tree.
+
+//! This module comprises S3 bucket READ ONLY access to download and parse
+//! Audit Proofs
+
+use anyhow::{bail, Result};
+use async_trait::async_trait;
+use aws_config::RetryConfig;
+use aws_sdk_s3 as s3;
+use clap::Args;
+use log::{debug, error};
+use std::convert::TryInto;
+use std::sync::Arc;
+use tokio::sync::RwLock;
+
+const MIN_BUCKET_CHARS: usize = 3;
+const MAX_BUCKET_CHARS: usize = 63;
+const ALLOWED_BUCKET_CHARS: [char; 38] = [
+    'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's',
+    't', 'u', 'v', 'w', 'x', 'y', 'z', '.', '-', '1', '2', '3', '4', '5', '6', '7', '8', '9', '0',
+];
```

Comment:
If you'd like to avoid typing all of this out, you can do something like this:
```rust
    let chars: Vec<char> = ('a'..='z')
        .into_iter()
        .chain(vec!['-', '.'].into_iter())
        .chain(('0'..='9').into_iter())
        .collect();
```

However, I don't believe the strategy above can be const. It does [generate the same set of chars we've manually added here](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=7835e98b72700257e082760a28ecb50d), so a bit more convenient. Your call!